src/main/java/net/snowflake/client/core/auth/oauth/TokenResponseDTO.java

```diff
@@ -0,0 +1,70 @@
+package net.snowflake.client.core.auth.oauth;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import net.snowflake.client.core.SnowflakeJdbcInternalApi;
+
+@SnowflakeJdbcInternalApi
+class TokenResponseDTO {
+
+  private final String accessToken;
+  private final String refreshToken;
+  private final String tokenType;
+  private final String scope;
+  private final String username;
+  private final boolean idpInitiated;
+  private final long expiresIn;
+  private final long refreshTokenExpiresIn;
+
+  @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
```

Comment:
What if some of these properties are missing? Will `nulls` be used?

Reply:
Yes